tests/e2e/helpers_e2e_test.go

```diff
@@ -92,3 +113,247 @@ func (s *IntegrationTestSuite) validateCLPosition(position model.Position, poolI
+	}
+
+	// System Under Test.
+	positionId, amount0, amount1, liquidity, poolIdLeaving, poolIdEntering, concentratedLockId := node.UnlockAndMigrateSharesToFullRangeConcentratedPosition(poolJoinAddress, fmt.Sprintf("%d", originalGammLockId), tokenOutMins.String(), coinsToMigrate.String())
+
+	s.validateMigrateResult(node, positionId, balancerPooId, poolIdLeaving, clPoolId, poolIdEntering, percentOfSharesToMigrate, liquidity, joinPoolAmt, amount0, amount1)
+
+	// If the lock was superfluid delegated, check the following
+	// - Correct amount remained delegated to the original Balancer intermediary account
+	// - Correct amount of Balancer pool shares remain in the original lock.
+	// - Correct amount is delegated to the cl intermediary account.
+	if superfluidDelegated && !superfluidUndelegating {
+		// If we migrated part of the shares:
+		// The intermediary account connection to the old gamm lock should still be present.
+		connectedIntermediaryAccount := node.QueryConnectedIntermediaryAccount(fmt.Sprintf("%d", originalGammLockId))
+		s.Require().Equal(balancerIntermediaryAcc.ValAddr, connectedIntermediaryAccount.ValAddr)
+
+		synthLock := node.QuerySyntheticLockupByLockupID(fmt.Sprintf("%d", originalGammLockId))
+		s.Require().Equal(synthLock.SynthDenom, fmt.Sprintf("%s/superbonding/%s", balancerLock.Coins[0].Denom, valAddr))
+
+		delegationResp, _ := node.QueryIntermediaryAccount(balancerIntermediaryAcc.Denom, chain.NodeConfigs[1].OperatorAddress)
+		delegation := delegationResp.GetDelegation()
+		// delegation amount remaining in the original balancer intermediary account should be equal to (original delegation amount - (original delegation amount * percent migrated))
+		s.Require().Equal(balancerDelegationPre.Shares.Sub(balancerDelegationPre.Shares.Mul(percentOfSharesToMigrate)).RoundInt().String(), delegation.Shares.RoundInt().String())
+
+		lock := node.QueryLockedById(fmt.Sprintf("%d", originalGammLockId))
+		// amount in original balancer lock should be equivalent to (total balancer share - coins migrated)
+		s.Require().Equal(totalBalancerPoolShare.Amount.Sub(coinsToMigrate.Amount).String(), lock.Coins[0].Amount.String())
+
+		// Check the new superfluid staked amount.
+		clIntermediaryAcc := node.QueryConnectedIntermediaryAccount(fmt.Sprintf("%d", concentratedLockId))
+		delegationResp, _ = node.QueryIntermediaryAccount(clIntermediaryAcc.Denom, chain.NodeConfigs[1].OperatorAddress)
+		delegation = delegationResp.GetDelegation()
+		s.Require().Equal(balancerDelegationPre.Shares.Mul(percentOfSharesToMigrate).RoundInt().String(), delegation.Shares.RoundInt().String())
+	}
+
+	// If the lock was superfluid undelegating
+	if superfluidDelegated && superfluidUndelegating {
+		// The intermediary account connection to the old gamm lock should be deleted.
+		connectedIntermediaryAccount := node.QueryConnectedIntermediaryAccount(fmt.Sprintf("%d", originalGammLockId))
+		s.Require().Equal(connectedIntermediaryAccount.Address, "")
+	}
```

Comment:
Can we do more detailed checks for the unbonding case? 
- amount left in lock
- amount delegated